### PR TITLE
Fixes  #35220 - CV Version Compare - add support for different content types

### PR DESCRIPTION
**Pull request by tazhibaevaaliya**
#### What are the changes introduced in this pull request?
Expanded CVVersionCompare to include other content types outside of just Packages
#### What are the testing steps for this pull request?
1. Select two version on the Content View Versions page
2. Hit "Compare" button
3. Make sure different content type tabs are rendered in the TableWrapper
4. Change the versions in the dropdowns
5. Ensure that the table is updates with new types of content types

### CODE REVIEW

**Comment by ianballou** [comment]
[test katello]

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareConfig.js
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareConfig.js:83]: Make this a URL just like version details config. We can do the same for content types that have a details view. The version details config should be a good pointer around which names should be a URL.
getProperty: item => (
          <a href={urlBuilder(`packages/${item?.id}`, '')}>
            {item?.nvrea}
          </a>),

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareConfig.js
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareConfig.js:130]: These columns should match the columns in version details config for this and other content types..

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareTable.js
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareTable.js:42]: String concat etc doesn't work in the translator. Let's leep the text simple like "No matching content found". That should be good enough.

**Comment by sjha4** [comment]
Some nitpicks. Looking great overall!

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js:160]: Could you hide this in this PR with a simple {false && ...} maybe since this is not yet implemented and this page will go into nightly as soon as merged.

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareTable.js
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareTable.js:42]: You can also use `<FormattedMessage>` if you want :yum:

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js
Comment (jeremylenz) [inline webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js:109]: Recommend adding `ouiaId` prop to this and all supported PF components.

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js:74]: however, `currentActiveKey` or `currentActiveTab` is a better name than just `tab`. So I recommend
```js
const [currentActiveTab, setCurrentActiveTab] = useState('RPM packages');
```
and then just use that everywhere.

File: webpack/scenes/ContentViews/Details/Versions/ContentViewVersions.js
Comment (jeremylenz) [inline webpack/scenes/ContentViews/Details/Versions/ContentViewVersions.js:145]:
```suggestion
        aria-label={`Select version ${version}`}
```

File: app/views/katello/api/v2/module_streams/compare.json.rabl
Comment (tazhibaevaaliya) [reply app/views/katello/api/v2/module_streams/compare.json.rabl:7]: @jeremylenz as for the counts, I initially had the counts for each content type tab. However, after looking at the mocks, noticed that there are no counts there. So, I removed them. Here is the link to the mocks, if you need it: https://marvelapp.com/prototype/6249jcd/screen/87503017

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js
Comment (tazhibaevaaliya) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js:89]: I think we decided to display the CVVesionCompareHeader and have the <Loading/> inside of the table because the last time I talked to @sjha4, he wanted to leave a chance for the users to change the version types in the drop down in case of there is a loading spinner. I hope this makes sense.

File: app/views/katello/api/v2/module_streams/compare.json.rabl
Comment (jeremylenz) [reply app/views/katello/api/v2/module_streams/compare.json.rabl:7]: Makes sense about the counts. I see that now on the mockups. thanks!

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js:89]: Yes, I'm just saying the logic here is confusing and hard to read. Can you maybe assign the logic to a variable instead, and then use that here?

So instead of 
```js
(((!(versionOneDetailsStatus === STATUS.PENDING
        ||
        versionTwoDetailsStatus === STATUS.PENDING || !(versionOne && versionTwo))) &&
```
you'd instead do
```js
// earlier in the component, before return
const showCompareTable = (!(versionOneDetailsStatus === STATUS.PENDING
        ||
        versionTwoDetailsStatus === STATUS.PENDING || !(versionOne && versionTwo));

// here
{showTabs && showCompareTable &&
```

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js:119]: I still don't understand why it's `|| <Loading />` here.

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js
Comment (tazhibaevaaliya) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js:89]: @jeremylenz sure. I will try to make the logic cleaner. Let me get back to you about it... Thanks

File: webpack/scenes/ContentViews/Details/ContentViewDetailActions.js
Comment (tazhibaevaaliya) [reply webpack/scenes/ContentViews/Details/ContentViewDetailActions.js:168]: @jeremylenz would that be "Docker tags" or "docker tags"? Would you mind confirming this with me?

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js
Comment (tazhibaevaaliya) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js:109]: @jeremylenz would you mind clarifying what do you mean by "PF"? :smiley:

File: webpack/scenes/ContentViews/Details/ContentViewDetailActions.js
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/ContentViewDetailActions.js:168]: "Docker tags" :+1:

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js:109]: sorry.. Patternfly components. The list of supported components is here: https://www.patternfly.org/v4/developer-resources/open-ui-automation#ouia-compliant-patternfly-4-components

File: app/views/katello/api/v2/module_streams/compare.json.rabl
Comment (tazhibaevaaliya) [reply app/views/katello/api/v2/module_streams/compare.json.rabl:7]: @jeremylenz @sjha4 would like to follow up on the comment about renaming " from comparison to appears_in". Should I make these changes?

Comment (sjha4) [reply app/views/katello/api/v2/module_streams/compare.json.rabl:7]: I am not strongly for or against either comparison or appears_in. My only concern would be changing it right away in case someone somewhere is using the API today. I don't mind adding a deprecation warning and changing it in the next release after some more feedback.

File: webpack/scenes/ContentViews/Details/ContentViewDetailActions.js
Comment (tazhibaevaaliya) [reply webpack/scenes/ContentViews/Details/ContentViewDetailActions.js:168]: @jeremylenz but then why in other cases, it would not be "Errata" instead of just "errata" or "Packages" and not "packages"?

File: app/views/katello/api/v2/module_streams/compare.json.rabl
Comment (tazhibaevaaliya) [reply app/views/katello/api/v2/module_streams/compare.json.rabl:7]: @sjha4 Sounds good. I will push my code with most of the things addressed in this PR except for this comment.

File: webpack/scenes/ContentViews/Details/ContentViewDetailActions.js
Comment (jeremylenz) [reply webpack/scenes/ContentViews/Details/ContentViewDetailActions.js]: It's just because Docker is a proper noun. https://www.docker.com/

**Comment by sjha4** [comment]
[test katello]

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompareHeader.js:102]: Translation..

File: webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js
Comment (sjha4) [inline webpack/scenes/ContentViews/Details/Versions/Compare/CVVersionCompare.js:50]: useState(__('RPM packages'));

**Review by sjha4 — COMMENTED** [review]
Some translation nitpicks..

**Comment by sjha4** [comment]
[test katello]

**Review by sjha4 — APPROVED** [review]
Nice work @tazhibaevaaliya! I am happy with the PR..Ack from my end.. 👍🏼 
I will let @jeremylenz merge this once he is happy..

**Review by jeremylenz — APPROVED** [review]
Works great.

Thanks for all the hard work @tazhibaevaaliya!

ACK :+1:

File: app/views/katello/api/v2/module_streams/compare.json.rabl
Comment (jeremylenz) [reply app/views/katello/api/v2/module_streams/compare.json.rabl:7]: Opened https://projects.theforeman.org/issues/35314 to track the rabl node change.

**Comment by tazhibaevaaliya** [comment]
@sjha4 @jeremylenz Thank you so much for all the support and guidance you have provided! I genuinely appreciate it 😄

**Pull request merged**
